Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1782,7 +1787,39 @@ def splunk_search_command(service):
 
     query = build_search_query(args)
     search_kwargs = build_search_kwargs(args)
-    search_job = service.jobs.create(query, **search_kwargs)  # type: ignore
+    job_sid = args.get("sid", None)
+    search_job = None
+
+    if not job_sid or not argToBoolean(args.get("polling", False)):
+        # create a new job to search the query.
+        search_job = service.jobs.create(query, **search_kwargs)  # type: ignore
+        job_sid = search_job["sid"]
+        args['sid'] = job_sid
+
+    status_cmd_result = splunk_job_status(service, args)
+    status = status_cmd_result.outputs['Status']
+    if status.lower() != 'done':
+        # Job is still running, schedule the next run of the command.
+        interval_in_secs = int(args.get('interval_in_seconds', 60))
+        polling_args = {
+            "sid": job_sid,
+            "polling": True,
+            "interval_in_seconds": interval_in_secs,
+        }
+        polling_args.update(args)
+        scheduled_command = ScheduledCommand(
+            command="splunk-search",
+            next_run_in_seconds=interval_in_secs,
+            args=polling_args,
+            timeout_in_seconds=600
+        )
+        status_cmd_result.scheduled_command = scheduled_command
+        status_cmd_result.readable_output = 'Job is still running, it may take a little while...'
+        return status_cmd_result
```

Comment:
Not sure I understand.
Won't this change the behavior of the regular search command?
The status will be `done` since it's a blocking search.

Reply:
I think what Guy meant is that for non-polling execution there is no need to call splunk_job_status().

Reply:
But still, the regular search behavior should not be changed in my opinion.
Please Correct me if I'm wrong.

Reply:
You're rigt, it shouldn't, just a redundant step we can skip if we aren't polling.

Reply:
Skipped the status section when it's not polling. Done.